src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -73,19 +80,19 @@ public String toString() {
 
     private static String buildBootstrapServers(final List<Broker> brokers) {
         final StringBuilder builder = new StringBuilder();
-        brokers.stream().forEach(broker -> builder.append(broker).append(","));
+        brokers.forEach(broker -> builder.append(broker).append(","));
         return builder.deleteCharAt(builder.length() - 1).toString();
     }
 
-    private Properties buildKafkaProperties(final List<Broker> brokers) {
+    private static Properties buildKafkaProperties(final List<Broker> brokers) {
         final Properties props = new Properties();
-        props.put("bootstrap.servers", buildBootstrapServers(brokers));
+        props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, buildBootstrapServers(brokers));
         return props;
     }
 
     @Scheduled(fixedDelay = 30000)
     private void updateBrokers() {
-        if (kafkaProperties != null) {
+        if (StringUtils.isEmpty(kafkaSettings.getBootstrapServers()) && kafkaProperties != null) {
```

Comment:
it means that you run a thread every 30 seconds even though the condition won't be fulfilled ever. could we change  it to have a scheduled action only if we need it?

Reply:
I've thought about `@ConditionalOnProperty` but I don't know if it works there. I'll give it a shot.

Reply:
Are you sure the scheduled task gets called with the current code ? KafkaLocationManager is not a bean (it used to) so it will not be scanned. I checked by adding a breakpoint and I'm never called. This looks like a bug...

Reply:
Yes, this is definitely a bug, `@Component` was gone during refactoring, when we started to using it without injections. could you please fix it (using scheduled thread pool will be fine with me) ? it will also help us to avoid unnecessary starts of thread when we start with bootstrap servers.
Thank you for catching it up !